src/commands/env/env-set.js

```diff
@@ -10,42 +14,64 @@ const { log, logJson, translateFromEnvelopeToMongo } = require('../../utils')
  * @returns {Promise<boolean>}
  */
 const envSet = async (key, value, options, command) => {
-  const { api, site } = command.netlify
+  const { context, scope } = options
+
+  const { api, cachedConfig, site } = command.netlify
   const siteId = site.id
 
   if (!siteId) {
     log('No site id found, please run inside a site folder or `netlify link`')
     return false
   }
 
-  const siteData = await api.getSite({ siteId })
+  const { siteInfo } = cachedConfig
+  let finalEnv
 
   // Get current environment variables set in the UI
-  const setInService = siteData.use_envelope ? setInEnvelope : setInMongo
-  const finalEnv = await setInService({ api, siteData, key, value })
+  if (siteInfo.use_envelope) {
+    finalEnv = await setInEnvelope({ api, siteInfo, key, value, context, scope })
+  } else if (context || scope) {
+    error(
+      `To specify a context or scope, please run ${chalk.yellow(
+        'netlify open:admin',
+      )} to open the Netlify UI and opt in to the new environment variables experience from Site settings`,
+    )
+    return false
+  } else {
+    finalEnv = await setInMongo({ api, siteInfo, key, value })
+  }
+
+  if (!finalEnv) {
+    return false
+  }
 
   // Return new environment variables of site if using json flag
   if (options.json) {
     logJson(finalEnv)
     return false
   }
 
-  log(`Set environment variable ${key}=${value} for site ${siteData.name}`)
+  const withScope = scope ? ` scoped to ${chalk.white(scope)}` : ''
+  log(
+    `Set environment variable ${chalk.yellow(`${key}${value ? '=' : ''}${value}`)}${withScope} in ${chalk.magenta(
```

Comment:
[sand] I think we're missing a `the` (and this will make it consistent with the phrasing for the other commands you added already)

```suggestion
    `Set environment variable ${chalk.yellow(`${key}${value ? '=' : ''}${value}`)}${withScope} in the ${chalk.magenta(
```